### PR TITLE
HARMONY-1355: Add UMM grid name parameter

### PR DESCRIPTION
**Pull request by vinnyinverso**
## Jira Issue ID
HARMONY-1355

## Description
Adds the new UMM grid name parameter.

## Local Test Steps
After activating your virtual environment, `pip install -e .` to get this branch version of harmony-py in your virtual environment.
Run through notebooks/examples/basic.ipynb. Modify the request to pass in a grid name. E.g.:
```
collection = Collection(id='C1234088182-EEDTEST')

request = Request(
    collection=collection,
    spatial=BBox(-165, 52, -140, 77),
    grid='LambertDurbin'
)
```
Inspect the request URL after submitting the job: https://harmony.uat.earthdata.nasa.gov/workflow-ui -- you should see `&grid=`.

Add an invalid grid name if you want to see an error.

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

**Review by chris-durbin — APPROVED** [review]
I'm thinking we may want to add an example for some of these new parameters in one of the Jupyter notebooks.

**Comment by vinnyinverso** [comment]
> I'm thinking we may want to add an example for some of these new parameters in one of the Jupyter notebooks.
In harmony and harmony-py @chris-durbin ?

**Review by indiejames — APPROVED** [review]

**Comment by vinnyinverso** [comment]
I added a UMM grid example to one of the notebooks per Chris's suggestion. We are going to change the name of the grid to LambertExample once the CMR grids update API is available. We may also re-ingest the example grid at a later date once the UMM grids support +/- latitude.

**Comment by vinnyinverso** [comment]
Added some notebook guidance for `destination_url` as well.

File: examples/s3_access.ipynb [hunk 1]
```diff
@@ -81,6 +83,7 @@
    "source": [
     "results = harmony_client.result_urls(job_id, link_type=LinkType.s3)\n",
     "print(results)\n",
+    "# NOTE: if you specified destination_url you'll have to retrieve your credentials in another manner\n",
```
Comment (asteiker) [inline examples/s3_access.ipynb:86]: What is that other manner? Could we add more explanation in markdown to offer a little more guidance? I would recommend we update intro_tutorial.ipynb as well - that notebook was intended to be the primary tutorial notebook when harmony-py was developed, as it provided more end-user guidance on all of the various parameters and access mechanisms (i.e. meant to be more of the comprehensive end-to-end tutorial, using production examples, etc.).

File: examples/s3_access.ipynb [hunk 2]
```diff
@@ -43,7 +43,9 @@
     "    crs='EPSG:3995',\n",
     "    format='image/tiff',\n",
     "    height=512,\n",
-    "    width=512\n",
+    "    width=512,\n",
+    "    # if desired, deliver results to a custom destination bucket\n",
```
Comment (asteiker) [inline examples/s3_access.ipynb:47]:
```suggestion
    "    # If desired, deliver results to a custom destination bucket. Note the bucket must reside in AWS us-west-2 region.\n",
```

File: examples/tutorial.ipynb [hunk 3]
```diff
@@ -166,11 +170,84 @@
...
     "    rasterio.plot.show(rasterio.open(r.result()))"
    ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "45a14473",
+   "metadata": {},
+   "source": [
+    "We can also specify a grid using the UMM grid name from the CMR. Let's build another request. This time, we'll request a specific granule and grid."
```
Comment (asteiker) [inline examples/tutorial.ipynb:180]: How does a user find what gird names are available? Could we include a CMR query that provides the associated grids available for that collection? And information on what those grid parameters are?

**Review by asteiker — COMMENTED** [review]
None of my comments are showstoppers for this particular PR but I had a few general thoughts: If our intention is to add guidance on our new features for end-users, I think we could add more "meat" when it comes to background on requirements for an s3 bucket destination, how to find and interpret grids associated with a give collection, etc. The tutorial.ipynb notebook only demonstrates examples in UAT which is okay but I don't think that's immediately clear to an end-user. The intro_tutorial.ipynb was written to be more of a comprehensive end-to-end notebook with "all" features included. But I know it has its own set of issues; it was written with a collection behind a EULA, it's not rendering from GitHub, and I'm sure I could identify other ways we could improve it. I also don't know how to verify whether the API changes are reflected in the documentation (https://harmony-py.readthedocs.io/en/latest/api.html) by default - this also may be broken?  But, these are issues we could address separately and worth more discussion.

File: examples/s3_access.ipynb (hunk 1 above)
Comment (vinnyinverso) [reply examples/s3_access.ipynb:86]: Added some further guidance. https://github.com/nasa/harmony-py/pull/55/commits/daaacd63f28ad7d4acc775c5b6a26d674ed7c056

File: examples/tutorial.ipynb (hunk 3 above)
Comment (vinnyinverso) [reply examples/tutorial.ipynb:180]: Sent a reply in Slack.

Comment (vinnyinverso) [reply examples/tutorial.ipynb:180]: Screen Shot 2023-02-24 at 2 59 44 PM

**Pull request merged**